src/main/kotlin/net/fabricmc/bot/extensions/NicknameExtension.kt

```diff
@@ -0,0 +1,56 @@
+package net.fabricmc.bot.extensions
+
+import com.kotlindiscord.kord.extensions.ExtensibleBot
+import com.kotlindiscord.kord.extensions.checks.inGuild
+import com.kotlindiscord.kord.extensions.extensions.Extension
+import dev.kord.core.behavior.edit
+import dev.kord.core.entity.Member
+import dev.kord.core.event.gateway.ReadyEvent
+import dev.kord.core.event.guild.MemberJoinEvent
+import dev.kord.core.event.guild.MemberUpdateEvent
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.filter
+import kotlinx.coroutines.flow.firstOrNull
+import net.fabricmc.bot.conf.config
+
+private val NAME_REGEX = Regex("^[!.\"+(-]")
```

Comment:
Aren't there other characters which are not listed here which can hoist, like `%`?